syntax = "proto3";

package imageclassificator;

// The request message containing the file to be classified.
message FileRequest {
    oneof file {
        bytes file_content = 1;
        string model_name = 2;
    }
}

// The response message containing the classification result.
message ClassificationResponse {
    string class_name = 1;
    float confidence = 2;
}

message ModelListResponse {
    repeated Model models = 1;
}

message Empty {}

message Model { 
    string name = 1;
    string description = 2;
    string version = 3;
    float accuracy = 4;
}

// The service definition.
service ImageClassificator {
    // Streams a file to the ML server and gets the classification result.
    rpc ClassifyFile(stream FileRequest) returns (ClassificationResponse);
    rpc ListModels(Empty) returns (ModelListResponse);
}